shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -136,6 +137,11 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
             if (currentBlockSize == timeoutMsg.ifStillBlockSize) {
                 breakFromAppend(out);
             }
+        } else if (msg instanceof Flush) {
+            Flush flush = (Flush) msg;
+            if (flush.getSegment().equals(segmentBeingAppendedTo) && flush.getWriterId().equals(writerIdPerformingAppends)) {
```

Comment:
Flush from other segment writers will not cause the client to to break from the current append.
We can simply this check to `if (flush.getWriterId().equals(writerIdPerformingAppends)) ` right?

Reply:
actually yes, but the session table maintained by the commandEncoder if tuple of (writerid , segmentName) , so i am maintaining the same here